functional-test/src/main/java/org/zanata/page/AbstractPage.java

```diff
@@ -435,4 +371,203 @@ private void assertReady(WebElement targetElement) {
         assertThat(targetElement.isDisplayed()).as("displayed").isTrue();
         assertThat(targetElement.isEnabled()).as("enabled").isTrue();
     }
+
+    /**
+     * Remove any visible notifications
+     */
+    public void removeNotifications() {
+        @SuppressWarnings("unchecked")
+        List<WebElement> notifications = (List<WebElement>) getExecutor()
+                .executeScript("return (typeof $ == 'undefined') ?  [] : " +
+                        "$('a.message__remove').toArray()");
+        if (notifications.isEmpty()) {
+            return;
+        }
+        log.info("Closing {} notifications", notifications.size());
+        for (WebElement notification : notifications) {
+            try {
+                notification.click();
+            } catch (WebDriverException exc) {
+                log.info("Missed a notification X click");
+            }
+        }
+        // Finally, forcibly un-is-active the message container - for speed
+        String script = "return (typeof $ == 'undefined') ?  [] : " +
```

Comment:
It is somewhat unfortunate that the Ajax silencer doesn't always wait for the notifications.